python/pyspark/sql/udf.py

```diff
@@ -465,6 +476,7 @@ def wrapper(*args: "ColumnOrName") -> Column:
         wrapper.returnType = self.returnType  # type: ignore[attr-defined]
         wrapper.evalType = self.evalType  # type: ignore[attr-defined]
         wrapper.deterministic = self.deterministic  # type: ignore[attr-defined]
+        wrapper.is_barrier = self.is_barrier  # type: ignore[attr-defined]
```

Comment:
I would keep this as an internal property by naming `_is_barrier`

Reply:
got it, let me rename it and add more comments